plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -154,53 +159,58 @@ private void checkInputReady()
 
     private synchronized List<ExchangeSourceHandle> createExchangeSourceHandles()
     {
-        Multimap<Integer, String> partitionFiles = ArrayListMultimap.create();
-        for (LocalFileSystemExchangeSinkHandle sinkHandle : finishedSinks) {
-            Path committedAttemptPath = getCommittedAttemptPath(sinkHandle);
-            Map<Integer, Path> partitions = getCommittedPartitions(committedAttemptPath);
-            partitions.forEach((partition, file) -> partitionFiles.put(partition, file.toAbsolutePath().toString()));
+        Multimap<Integer, URI> partitionFiles = ArrayListMultimap.create();
+        for (Integer partitionId : finishedPartitions) {
```

Comment:
nit: `int`

Reply:
I would expect it to be handled by the JVM and not do unnecessary work - but not sure.